source/CIS/cis-6.rst

```diff
@@ -0,0 +1,118 @@
+
+- *ItemCreatedEvent* loggs the ``item_id``, ``metadata_url``, and the ``initial_status``;
+- *ItemStatusChangedEvent* loggs the ``item_id``, ``new_status``, and the ``additional_data``;
+
+Specification
+=============
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in :rfc:`2119`.
+
+
+General types and serialization
+-------------------------------
+
+.. _CIS-6-ItemId:
+
+``ItemId``
+^^^^^^^^^^
+
+Each item MUST have a unique ``ItemId``.
+An ``ItemId`` is a variable-length ASCII string up to 255 characters.
+
+The identifier is serialized as: first 1 byte for the length followed this many bytes for the ASCII encoding of the identifier::
+
+  StandardIdentifier ::= (n: Byte) (id: Byteⁿ)
+
+.. _CIS-6-MetadataUrl:
+
+``MetadataUrl``
+^^^^^^^^^^^^^^^
+
+A URL and optional checksum for metadata stored outside of this contract.
+
+It is serialized as: 2 bytes for the length (``n``) of the metadata URL in little-endian and then this many bytes for the URL to the metadata (``url``) followed by an optional checksum.
+The checksum is serialized by 1 byte to indicate whether a hash of the metadata is included.
+If its value is 0, then there is no hash; if the value is 1, then 32 bytes for a SHA256 hash (``hash``) follows::
+
+  MetadataChecksum ::= (0: Byte)
+                     | (1: Byte) (hash: Byte³²)
+
+  MetadataUrl ::= (n: Byte²) (url: Byteⁿ) (checksum: MetadataChecksum)
+
+.. _CIS-6-Status:
+
+``Status``
+^^^^^^^^^^
+
+The status defined by this specification are serialized using one byte to discriminate the different statuses that an item can have.
+The smart contract can have up to 255 different statuses defined for an item.
+
+It is serialized as: a byte of value (``n``)::
+
+  Status ::= (n: Byte)
+
+.. _CIS-6-AdditionalData:
+
+``AdditionalData``
+^^^^^^^^^^^^^^^^^^
+
+Additional bytes to include in the event ``ItemStatusChangedEvent``, which can be used to add additional usecase-specific data (e.g. temperature, longitude, latitude, ...) to the event.
+
+It is serialized as: the first 2 bytes encode the length (``n``) of the data, followed by this many bytes for the data (``data``)::
+
+  AdditionalData ::= (n: Byte²) (data: Byteⁿ)
+
+Logged events
+-------------
+
+The events defined by this specification are serialized using one byte to discriminate the different events.
+A custom event SHOULD NOT have a first byte colliding with any of the events defined by this specification.
+
+.. _CIS-6-events-ItemCreatedEvent:
+
+``ItemCreatedEvent``
+^^^^^^^^^^^^^^^^^^^^
+
+A ``ItemCreatedEvent`` event MUST be logged when a new item is created in the smart contract.
+
+The ``ItemCreatedEvent`` event is serialized as: first a byte with the value of 0, followed by the :ref:`CIS-6-ItemId` (``id``), the :ref:`CIS-6-MetadataUrl` (``metadata``), and then the :ref:`CIS-6-Status` (``initial_status``)::
```

Comment:
We are using `255 - n` for the other event ids (CIS-2, CIS-4, etc.). Why are we not doing it here?

Reply:
Makes sense:

Then CIS-5 should be updated to use the NonceEvent tag 250 and 243, 242, 241, 240, 239, 238 for the remaining 6 events.
And this CIS-6 standard will use the tags 237 and 236.